### ISSUE TITLE
Selection Column has no aria-label for screen reader a11y

### ISSUE THREAD

**Issue by httpete** [mui/mui-x#14285]
### Search keywords

keyboard a11y columns

### Latest version

- [X] I have tested the latest version

### Summary

If you look at the markup rendered in the empty cell / select all header cell:

``

<div class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-withBorderColor MuiDataGrid-columnHeaderCheckbox" role="columnheader" tabindex="-1" aria-colindex="1" aria-sort="none" data-field="__check__" style="height: 39px; width: 50px; min-width: 50px; max-width: 50px;">stuff in here</div>

```

It has no aria-label attribute on it. I need it to have this in order to have the screen reader announce what the cell is when focused.:

```

aria-label="Selection column"
```


### Examples

_No response_

### Motivation

We have extensive screen reader a11y requirements and this one small addition would help that rating .

### Order ID or Support key 💳 (optional)

79357

**Comment by MBilalShafi** [comment]
@httpete May I know which screen reader are you using?

Mac VoiceOver _does_ announce `Select all rows` (`aria-label` applied on the `input` checkbox component for "Select All" checkbox)

https://github.com/user-attachments/assets/0c4c22cb-657b-4497-af3e-973a2bd7b6d6

**Comment by httpete** [comment]
But in the case of single select, it does not.

**Comment by michelengelen** [comment]
Hey @httpete ... could you share a minimal reproduction for this?
From your screenshot it seems you are using [Single row selection](https://mui.com/x/react-data-grid/row-selection/#single-row-selection) with a custom radio column. Is that correct?

**Comment by httpete** [comment]
I do override the baseCheckbox slot with a Radio in case of single select, but all that is needed to simulate the above is:
```
     disableMultipleRowSelection={true}
     ```
on the dataGrid, (it hides the select all checkbox in that case)  my overriding the radio has no bearing on this.

**Comment by michelengelen** [comment]
OK, I did manage to find the culprit.

https://github.com/mui/mui-x/blob/45b28354e18d4178e3ebe2fa33e89fca6e4221dd/packages/x-data-grid/src/components/columnHeaders/GridGenericColumnHeaderItem.tsx#L114

Here we do not set the `aria-label` property because there is a custom component being rendered. 
Otherwise you could just use the `colDef.headerName` property to change that label.

I'll add it to the board, since we cannot just remove that condition. It would be a breaking change!
